### PR TITLE
fix: `createQuery` tests part 2

### PR DESCRIPTION
**Pull request by jennyckaplan**
Continues fixing the `createQuery` tests.

### CODE REVIEW

File: packages/solid-query/src/__tests__/createQuery.test.tsx [hunk 1]
```diff
@@ -1199,21 +1203,24 @@ describe('createQuery', () => {
...
+      const [, setRenderValue] = createSignal({})
+
+      const rerender = () => setRenderValue({})
+
       const state = createQuery<string, Error>(
         key,
         () => (runs === 0 ? 'test' : 'test2'),
         {
-          select: NotReact.useCallback(() => {
+          select: () => {
```
Comment (lukesmurray) [inline packages/solid-query/src/__tests__/createQuery.test.tsx:1220]: are we missing get here?

**Review by lukesmurray — APPROVED** [review]

File: packages/solid-query/src/__tests__/createQuery.test.tsx [hunk 2]
```diff
@@ -2601,35 +2460,6 @@ describe('createQuery', () => {
     expect(states[1]).toMatchObject({ status: 'success' })
   })
 
-  it('should batch re-renders', async () => {
```
Comment (lukesmurray) [inline packages/solid-query/src/__tests__/createQuery.test.tsx:2604]: do we want to delete tests or just skip them?

Comment (jennyckaplan) [reply packages/solid-query/src/__tests__/createQuery.test.tsx:2604]: I think delete if they are React specific tests. Unless we want to `skip` and then go through them together to make sure there's not a similar solid edge case, but anything regarding re-renders doesn't make sense in the Solid context

File: packages/solid-query/src/__tests__/createQuery.test.tsx [hunk 3]
```diff
@@ -1419,7 +1276,7 @@ describe('createQuery', () => {
     // Fetched
     expect(states[1]).toMatchObject({ data: 1 })
     // Switch
-    expect(states[2]).toMatchObject({ data: undefined, dataUpdatedAt: 0 })
```
Comment (jennyckaplan) [inline packages/solid-query/src/__tests__/createQuery.test.tsx:1422]: `data` isn't explicitly set to `undefined` here (it's just missing from the state object). Is this expected / fine or a bug?

**Review by lukesmurray — APPROVED** [review]
